### DISCUSSION TITLE
Ephemeris date limits?

### DISCUSSION THREAD

**Discussion by Paul-Gilmartin** [Stellarium/stellarium#3099] [Q&A]
I'm trying to create an ephemeris for 2023 DW.  It seems that whatever end date I enter, the calculation ends at 2023-07-09.  Is this a hard limit?  Elements from ssystem.ini are:
[2023dw]
orbit_Inclination = 5.81541
orbit_AscendingNode = 326.14525
radius = 1
albedo = 0.15
coord_func = comet_orbit
orbit_SemiMajorAxis = 0.8197942
orbit_visualization_period = 271.11633746248054
slope_parameter = 0.15
orbit_MeanMotion = 1.32784337
absolute_magnitude = 24.32
orbit_MeanAnomaly = 120.0172
type = asteroid
name = 2023 DW
orbit_Epoch = 2460000.5
orbit_ArgOfPericenter = 40.44512
orbit_Eccentricity = 0.3966995

Thanks,
gil

**Comment by gzotti** [comment]
No problem with some other objects here. Can you show a screenshot of the AstroCalc panel after creating the ephemeris?

**Reply by Paul-Gilmartin** [reply]
Yes, thanks.  And I neglected to specify: Stellarium 1.2 on MacBook Pro;
macOS 12.6.3 (21G419).  As a control, I tried Moon for month of July amd
got all 31 days.  I changed *only* Moon do 2023 DW and got only 9 days.
Thanks,
gil

**Comment by Paul-Gilmartin** [comment]
<img width="871" alt="Screen Shot 2023-03-13 at 18 13 08" src="https://user-images.githubusercontent.com/13041488/224860532-a9d42db0-d4e3-4f9f-83c4-be2be8fac0e4.png">

**Reply by gzotti** [reply]
OK, strange. @alex-w does ephemeris stop when magnitude is too low? Other ideas?

**Reply by alex-w** [reply]
No, this is not related to magnitude. I fear it related to valid range of orbital data.

**Comment by Paul-Gilmartin** [comment]
I got interested when I saw news reports that 2023 DW will make a close approch
on 2046-02-14.  Around then, the Stellarium main view shows it (or its position) crossing the sky rapidly.
Thanks,
gil

**Comment by Paul-Gilmartin** [comment]
Should the elements show the "valid range"?  I set the time to 2046-02-24 21:37:22
and see:
2023 DW
Type: asteroid
Magnitude: 14.69
Absolute Magnitude: 24.32
RA/Dec (J2000.0): 13h27m42.14s/-10°43'04.2"
RA/Dec (on date): 13h30m09.22s/-10°57'27.1"
HA/Dec: 17h47m35.25s/-10°57'27.1" (apparent)
Az./Alt.: +96°29'00.2"/-9°22'35.9" (apparent)
Gal. long./lat.: +317°13'10.9"/+51°08'42.4"
Supergal. long./lat.: +129°01'08.3"/+4°49'25.1"
Ecl. long./lat. (J2000.0): +204°14'58.9"/-1°24'52.5"
Ecl. long./lat. (on date): +204°53'51.6"/-1°25'03.1"
Ecliptic obliquity (on date): +23°26'06.2"
Mean Sidereal Time: 7h17m43.7s
Apparent Sidereal Time: 7h17m44.5s
Rise: 22h16m
Transit: 3h02m
Set: 7h41m
Parallactic Angle: -50°48'53.9"
IAU Constellation: Vir
Hourly motion: +3°10'01" towards 277.1°
Hourly motion: dα=-3°12'00" dδ=+0°24'03"
Elongation: 121°44'56.8"
Elong. in Ecl.Long.: W121°45'04"
Phase angle: +58°01'33.4"
Distance from Sun: 0.990 AU (148.112 M km)
Distance: 0.004572 AU (683922.526 km)
Light time: 0h00m02.3s 
Orbital velocity: 26.644 km/s
Sidereal period: 271.12 days (0.742 a)
Apparent diameter: +0°00'00.60"
Diameter: 2.0 km
Solar Az./Alt.: +297°05'32"/-45°52'15"
Lunar Az./Alt.: +262°26'32"/+51°41'07"
NOTE: orbital elements outdated -- consider updating!

Thanks,
gil

**Comment by Paul-Gilmartin** [comment]
Aha!  The line "NOTE: orbital elements outdated -- consider updating!" appears
exactly at July 10.

Thanks,
gil

**Reply by gzotti** [reply]
Ah yes. I was not aware that AstroCalc then does not provide output. 
You can circumvent the test for meaningful output by adding `orbit_good=0` to the elements. As always, see Appendix D in the User Guide for details.

**Reply by alex-w** [reply]
OK, I disabled the checking in AstroCalc now

**Reply by gzotti** [reply]
It may be wise to add a boolean column "elements out of range" or so. But we need a short column title. Or yet another checkbox to switch the test? "Ignore date test",  Tooltip: "Allow results outside recommended range of orbital elements". I can write a paragraph for the SUG what this means.

**Reply by 10110111** [reply]
A short title: "Valid" (with inverted meaning).

**Reply by alex-w** [reply]
@gzotti please check AstroCalc/Ephemeris tool in the master

**Reply by gzotti** [reply]
Looks almost good. What does IDT mean? (Ignore D???.T???) IMHO there is enough space to show slightly longer abbreviations. Also Hor.Coord.

**Reply by gzotti** [reply]
Being blind a bit, IDT seems to abbreviate my own suggestion... :-D 
Better: "Useful dates only". I am afraid there is no 1-word label for this concept. (And tooltip: "Allow results outside recommended date range of orbital elements"

**Comment by Paul-Gilmartin** [comment]
Thanks.  orbit_good=0 did what I was trying.
Perhaps the user-friendly change in the Ephemeris panel would be a single
warning line that the user-specified limits exceed "orbit_good", which could
readily be found by a naive user in the User Guide.
Perhaps a checkbox:  "[ ] ignore orbit_good".

Thanks again,
gil
